### PR TITLE
consts moved to their own file

### PR DESCRIPTION
**Pull request by stevencartavia**
Time spent on this PR:  

Resolves: #1196 

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

# Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: indexer/src/types/header.ts [hunk 1]
```diff
@@ -15,7 +18,7 @@ import {
 import { KAKAROT } from "../provider.ts";
 
 // A default block gas limit in case the call to get_block_gas_limit fails.
-const DEFAULT_BLOCK_GAS_LIMIT = Deno.env.get("DEFAULT_BLOCK_GAS_LIMIT");
+
 if (DEFAULT_BLOCK_GAS_LIMIT === undefined) {
   throw new Error("ENV: DEFAULT_BLOCK_GAS_LIMIT is not set");
 }
```
Comment (tcoratger) [inline indexer/src/types/header.ts:24]: Can we move that also to the constant file?

File: indexer/src/utils/filter.ts [hunk 2]
```diff
@@ -1,7 +1,9 @@
 // Starknet
 import { Event, Transaction, TransactionReceipt } from "../deps.ts";
 
-const KAKAROT_ADDRESS = Deno.env.get("KAKAROT_ADDRESS");
+// Constants
+import { KAKAROT_ADDRESS } from "../constants.ts";
+
 if (KAKAROT_ADDRESS === undefined) {
   throw new Error("ENV: KAKAROT_ADDRESS is not set");
 }
```
Comment (tcoratger) [inline indexer/src/utils/filter.ts:9]: Can we move that also to the constant file?

**Comment by stevencartavia** [comment]
for some reason the variable RPC_URL is not returning the desired output in the constants file, I tried different things but I didn't get it yet

File: indexer/src/main.ts [hunk 3]
```diff
@@ -34,15 +43,10 @@ import {
...
-const STREAM_URL = Deno.env.get("STREAM_URL") ?? "http://localhost:7171";
-const STARTING_BLOCK = Number(Deno.env.get("STARTING_BLOCK")) ?? 0;
 if (!Number.isSafeInteger(STARTING_BLOCK) || STARTING_BLOCK < 0) {
   throw new Error("Invalid STARTING_BLOCK");
 }
-const SINK_TYPE = Deno.env.get("SINK_TYPE") ?? "console";
+
 if (SINK_TYPE !== "console" && SINK_TYPE !== "mongo") {
   throw new Error("Invalid SINK_TYPE");
 }
```
Comment (tcoratger) [inline indexer/src/main.ts:52]: Can we also migrate this? (everything related to constants/check of constants should be in the constant file ideally)

File: indexer/src/main.ts [hunk 4]
```diff
@@ -34,15 +43,10 @@ import {
...
 import { Bloom, Trie } from "./deps.ts";
 
-const AUTH_TOKEN = Deno.env.get("APIBARA_AUTH_TOKEN") ?? "";
-const TRANSACTION_EXECUTED = hash.getSelectorFromName("transaction_executed");
-
-const STREAM_URL = Deno.env.get("STREAM_URL") ?? "http://localhost:7171";
-const STARTING_BLOCK = Number(Deno.env.get("STARTING_BLOCK")) ?? 0;
 if (!Number.isSafeInteger(STARTING_BLOCK) || STARTING_BLOCK < 0) {
   throw new Error("Invalid STARTING_BLOCK");
 }
```
Comment (tcoratger) [inline indexer/src/main.ts:48]: Can we also migrate this?

File: indexer/src/constants.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+export const AUTH_TOKEN = Deno.env.get("APIBARA_AUTH_TOKEN") ?? "";
+
+// Get stream URL or returns "http://localhost:7171" if the value is null or undefined
+export const STREAM_URL = Deno.env.get("STREAM_URL") ?? "http://localhost:7171";
+
+// Get the starting block or returns 0 if the value is null or undefined
+export const STARTING_BLOCK = setTimeout(() => {
+    const addr = Number(Deno.env.get("STARTING_BLOCK")) ?? 0;
+    if (!Number.isSafeInteger(addr) || addr < 0) throw new Error("Invalid STARTING_BLOCK");
+    return addr;
```
Comment (tcoratger) [inline indexer/src/constants.ts:21]:
```suggestion
    const startingBlock = Number(Deno.env.get("STARTING_BLOCK")) ?? 0;
    if (!Number.isSafeInteger(startingBlock) || startingBlock < 0) throw new Error("Invalid STARTING_BLOCK");
    return startingBlock;
```

**Comment by tcoratger** [comment]
@stevencartavia kakarot submodule should not be updated here can you revert?

**Review by tcoratger — APPROVED** [review]

**Pull request merged**